File: Linked List/0203.Remove Linked List Elements/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        ans = ListNode()
        while head and head.val == val:
            head = head.next
        ans = head
        while head and head.next:
            if head.next.val == val:
                head.next = head.next.next
                continue
            head = head.next
        return ans

File: Linked List/0203.Remove Linked List Elements/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode ans = new ListNode();
        while (head != null && head.val == val) {
            head = head.next;
        }
        ans = head;
        while (head != null && head.next != null) {
            if (head.next.val == val) {
                head.next = head.next.next;
                continue;
            }
            head = head.next;
        }
        return ans;
    }
}